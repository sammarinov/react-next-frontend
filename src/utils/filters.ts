import { IFeatured, IArticle } from "@Redux/app/type";

export const filterFeatured = (featured: Array<IFeatured>, limit: number = -1) => {
    featured.sort((a, b) => {
        if (a.topFeatured && !b.topFeatured) return 1;
        if (!a.topFeatured && b.topFeatured) return -1;
        return parseInt(a.price) - parseInt(b.price);
    })

    if (limit > 0) return featured.slice(0, limit);
    return featured;
}

export const filterArticles = (articles: Array<IArticle>, limit: number = -1) => {
    articles.sort((a, b) => {
        return Date.parse(a.created) - Date.parse(b.created);
    })

    if (limit > 0) return articles.slice(0, limit);
    return articles;
}