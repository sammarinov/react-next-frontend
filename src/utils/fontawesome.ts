import { library } from '@fortawesome/fontawesome-svg-core';

import {
    faArrowRight
} from '@fortawesome/free-solid-svg-icons';

import {
    faTwitter,
    faFacebook,
    faInstagram,
    faYoutube
} from '@fortawesome/free-brands-svg-icons';

library.add(
    faArrowRight,
    faTwitter,
    faFacebook,
    faInstagram,
    faYoutube
)