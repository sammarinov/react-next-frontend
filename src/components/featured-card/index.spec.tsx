import * as React from 'react';

import { render } from '@Test/utils';
import FeaturedCard from '.';

describe('FeaturedCard', () => {
    it('should match snapshot', () => {
        const { container } = render(<FeaturedCard item={
            {
                "id": 11,
                "caption": "Mickey Mouse",
                "description": "Micke Mouse is a cartoon character and the mascot of The Walt Disney Company. He was created by Walt Disney and Ub Iwerks at the Walt Disney Studios in 1928. An anthropomorphic mouse who typically wears red shorts, large yellow shoes, and white gloves, Mickey is one of the world's most recognizable characters.",
                "picture": "mickey.png",
                "favorite": "68",
                "price": "99.99",
                "actions": ["ADD_TO_CART", "MORE_DETAILS"],
                "topFeatured": true
            }
        } />);

        expect(container).toMatchSnapshot();
    });
});
