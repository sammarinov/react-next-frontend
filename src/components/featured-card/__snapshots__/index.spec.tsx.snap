// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedCard should match snapshot 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  min-width: 1.25rem;
  min-height: 1.25rem;
  border: 1px solid transparent;
  outline: none;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.3125rem;
  grid-template-columns: auto 3.125rem;
  cursor: pointer;
  border-radius: 0.9375rem;
  background-color: #ffffff;
  border-color: #102027;
  color: #102027;
}

.c0:focus {
  outline: none;
}

<div>
  <div
    class="featured-card "
  >
    <div
      class="featured-card-caption"
    >
       
      Mickey Mouse
       
    </div>
    <div
      class="featured-card-description"
    >
       
      Micke Mouse is a cartoon character and the mascot of The Walt Disney Company. He was created by Walt Disney and Ub Iwerks at the Walt Disney Studios in 1928. An anthropomorphic mouse who typically wears red shorts, large yellow shoes, and white gloves, Mickey is one of the world's most recognizable characters.
       
    </div>
    <img
      class="featured-card-image"
      src="images/mickey.png"
    />
    <div
      class="featured-card-favorite"
    >
      <img
        class="featured-card-favorite-icon"
        src="images/love.png"
      />
      <div
        class="featured-card-favorite-bar-container"
      >
        <div
          class="undefined favorite-bar-container"
        >
          <div
            class="favorite-bar"
            style="width: 68%;"
          />
        </div>
      </div>
      <div
        class="featured-card-favorite-value"
      >
        68
        %
      </div>
    </div>
    <div
      class="featured-card-price"
    >
      99.99
       
      <span
        class="featured-card-price-comment"
      >
        inc VAT
      </span>
    </div>
    <div
      class="featured-card-actions"
    >
      <div
        class="featured-card-actions-item"
      >
        <button
          class="c0"
        >
          <span>
            translations:ADD_TO_CART
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-right fa-w-14 "
            data-icon="arrow-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="featured-card-actions-item"
      >
        <button
          class="c0"
        >
          <span>
            translations:MORE_DETAILS
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-right fa-w-14 "
            data-icon="arrow-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
